### PR TITLE
Move linear gradient animations into timeline

### PR DESCRIPTION
**Pull request by Owen-Delisle**

### CODE REVIEW

**Comment by jbeenie** [comment]
can you post some sample code please

**Comment by Owen-Delisle** [comment]
[Contraction.zip](https://github.com/SlantDesign/FlowTemplates/files/4766468/Contraction.zip)

File: Packages/Web/templates/html/fillGradient.html.flt [hunk 1]
```diff
@@ -1,3 +1,2 @@
 <defs id="<%= id %>-defs">
-<%- insert("svg/gradient.svg.flt", from: Web, indentation: indentation + 1) %>
 </defs>
```
Comment (jbeenie) [inline Packages/Web/templates/html/fillGradient.html.flt:2]: do we still use this template file?

Comment (Owen-Delisle) [reply Packages/Web/templates/html/fillGradient.html.flt:2]: Yes we use it in the Timeline.

**Comment by jbeenie** [comment]
Can you post an updated exported sample project showing the code?

**Comment by Owen-Delisle** [comment]
[MoveStops.zip](https://github.com/SlantDesign/FlowTemplates/files/4773169/MoveStops.zip)

File: Packages/Web/lib/util.flp [hunk 2]
```diff
@@ -427,6 +427,12 @@ func svgAnimationValue(attributeName, from keyvalue)
   case "stroke-width":
     return keyvalue.value
 
+  case "fillGradient.stop.position":
+    return keyvalue.value
+
+  case "stop":
+    return keyvalue.position
```
Comment (jbeenie) [inline Packages/Web/lib/util.flp:434]: these two cases look redundant?

Comment (jbeenie) [reply Packages/Web/lib/util.flp:434]: does keyvalue have a field called "position"?

File: Packages/Web/lib/util.flp [hunk 3]
```diff
@@ -477,3 +480,27 @@ func trackValues(property, _ keyvalues, layer, imageFolderPath, importImage, _ d
...
+    for gradient in gradients
+        var stop = gradient.stops[stopIndex]
+        values = values.appending(stop.position)
+    end
+
+    if values.areIdentical()
+        return nil
+    end
+
+    return ["values": values, "keytimes": keytimes, "keysplines": keysplines]
```
Comment (jbeenie) [inline Packages/Web/lib/util.flp:505]: just return the stop values from this function

File: Packages/Web/templates/svg/stop.svg.flt [hunk 4]
```diff
@@ -1 +1,14 @@
-<stop class="stop<%= i %>" offset="<%= stop.position * 100 %>%" stop-color="<%= stop.color.hex %>" ></stop>
\ No newline at end of file
+<stop class="stop<%= i %>" offset="<%= stop.position * 100 %>%" stop-color="<%= stop.color.hex %>" >
+    <%-
+    if stopAnimations
+        var stopAnimation = Web.gradientStopAnimation(timeline: timeline, layer: layer, stopIndex: i, property: "fillgradient.stop.position", indentation: 3)
+        if stopAnimation != nil
+            var values = stopAnimation["values"]
+            var keytimes = stopAnimation["keytimes"]
+            var keysplines = stopAnimation["keysplines"]
```
Comment (jbeenie) [inline Packages/Web/templates/svg/stop.svg.flt:8]: grab the key splines from the track

File: Packages/Web/templates/svg/stop.svg.flt [hunk 5]
```diff
@@ -1 +1,14 @@
-<stop class="stop<%= i %>" offset="<%= stop.position * 100 %>%" stop-color="<%= stop.color.hex %>" ></stop>
\ No newline at end of file
+<stop class="stop<%= i %>" offset="<%= stop.position * 100 %>%" stop-color="<%= stop.color.hex %>" >
+    <%-
+    if stopAnimations
+        var stopAnimation = Web.gradientStopAnimation(timeline: timeline, layer: layer, stopIndex: i, property: "fillgradient.stop.position", indentation: 3)
+        if stopAnimation != nil
+            var values = stopAnimation["values"]
+            var keytimes = stopAnimation["keytimes"]
```
Comment (jbeenie) [inline Packages/Web/templates/svg/stop.svg.flt:7]: grab the key times from the track

File: Packages/Web/templates/svg/stop.svg.flt [hunk 6]
```diff
@@ -1 +1,18 @@
-<stop class="stop<%= i %>" offset="<%= stop.position * 100 %>%" stop-color="<%= stop.color.hex %>" ></stop>
\ No newline at end of file
+<%-
+var track = timeline.trackFor(layer: layer, property: "fillGradient")
+-%>
+
+<stop class="stop<%= i %>" offset="<%= stop.position * 100 %>%" stop-color="<%= stop.color.hex %>" >
+    <%-
+    if stopAnimations
+        var stopValues = Web.gradientStopValues(timeline: timeline, track: track, layer: layer, stopIndex: i, property: "fillgradient.stop.position", indentation: 3)
```
Comment (jbeenie) [inline Packages/Web/templates/svg/stop.svg.flt:8]: lets put: `var track = timeline.trackFor(layer: layer, property: "fillGradient")`

here

**Pull request merged**